src/merge_events.cpp

```diff
@@ -0,0 +1,205 @@
+#include <set>
+#include <vector>
+
+#include <backends.hpp>
+#include <cell_group.hpp>
+#include <cell_group_factory.hpp>
+#include <domain_decomposition.hpp>
+#include <merge_events.hpp>
+#include <model.hpp>
+#include <recipe.hpp>
+#include <util/filter.hpp>
+#include <util/span.hpp>
+#include <util/unique_any.hpp>
+#include <profiling/profiler.hpp>
+
+namespace arb {
+
+namespace impl {
+
+// The tournament tree data structure is used to merge k sorted lists of events.
+// See online for high-level information about tournament trees.
```

Comment:
Would this be a good link to add (google I feel lucky)
http://www.geeksforgeeks.org/tournament-tree-and-binary-heap/

Reply:
I thought about adding that exact link! But decided against it because links can go stale over time.
When googling "tournament tree" the first few hits are good references, so I figure folks in the future will just google it for themselves.
Keep in mind that this is a low level part of the library, designed just for the task in this file.

Reply:
https://web.archive.org/web/20170611232149/http://www.geeksforgeeks.org/tournament-tree-and-binary-heap/